src/instructlab/model/evaluate.py

```diff
@@ -33,889 +32,768 @@ class Benchmark(str, enum.Enum):
+                        model_name=m_path,
+                        max_workers=None,
+                        gpus=gpus,
+                        backend=backend,
+                        enable_serving_output=enable_serving_output,
+                    )
+                    overall_score, individual_scores = evaluator.run(api_base)
+                    overall_scores.append(overall_score)
+                    individual_scores_list.append(individual_scores)
+                finally:
+                    if server is not None:
+                        server.shutdown()
 
-        gpus, effective_gpus = get_gpus(eval_serve, gpus)
-        if gpus:
-            tps_prefix = "--tensor-parallel-size"
-            if contains_argument(tps_prefix, eval_serve.vllm.vllm_args):
-                click.secho(
-                    "Using gpus from --gpus or config and ignoring --tensor-parallel-size configured in serve vllm_args",
-                    fg="yellow",
-                )
-            eval_serve.vllm.vllm_args.extend([tps_prefix, str(gpus)])
-        elif effective_gpus < 1:
-            click.secho(
-                "Evaluate is currently not configured to use GPUs. If you are on a GPU-enabled system edit your config or pass the number of GPUs you would like to use with '--gpus'",
-                fg="yellow",
-            )
+            overall_score = overall_scores[0]
+            base_overall_score = overall_scores[1]
+            individual_scores = individual_scores_list[0]
+            base_individual_scores = individual_scores_list[1]
 
-        if max_workers is not None and isinstance(max_workers, int):
-            # Recommend max-workers based on hardware configuration: min(#GPUs being used * 10, #CPU cores) +- 50%
-            # Edge cases:
-            # - Many GPUs, not many CPUs: Unlikely, workers might not be able to keep the GPUs busy but recommendation can be ignored.
-            # - Many CPUs, not many GPUs: More likely, 10 workers per GPU should still be reasonable.
-            target_max_workers = min(max(effective_gpus, 1) * 10, get_cpu_count())
-            recommended_min_workers = max(target_max_workers // 2, 1)
-            recommended_max_workers = max(int(target_max_workers // 0.5), 1)
-            if (
-                max_workers > recommended_max_workers
-                or max_workers < recommended_min_workers
-            ):
-                logger.warning(
-                    f"Based on your hardware configuration, when using vLLM, we recommend setting max-workers between {recommended_min_workers} and {recommended_max_workers} for optimal performance"
-                )
-    elif backend == backends.LLAMA_CPP:
-        if eval_serve.llama_cpp.max_ctx_size < 5120:
-            eval_serve.llama_cpp.max_ctx_size = 5120
-            logger.debug(
-                "Evaluate requires a context size of >= 5120, ignoring serve configuration for max_ctx_size"
+            print("# KNOWLEDGE EVALUATION REPORT\n")
```

Comment:
logger if possible so we can capture it. If this one is an exception for formatting purposes that's fine!

Reply:
@RobotSail wdyt?

Reply:
https://github.com/instructlab/instructlab/issues/2900

Reply:
Thanks, Oleg!